inductiva/resources/machines.py

```diff
@@ -92,37 +88,37 @@ def terminate(self):
                     num_instances=self.num_machines,
                     spot=self.spot,
                     disk_size_gb=self.disk_size_gb,
-                    zone=self.zone,
-                )
-            try:
-                logging.info("Terminating machine group."
-                             "This may take a few minutes.")
-                start_time = time.time()
-                api_instance.delete_instance_group(body=instance_group_config)
-                termination_time_mins = (time.time() - start_time) / 60
-                logging.info(
-                    "Machine group of %s machines successfully "
-                    "terminated in %.2f mins.", self.num_machines,
-                    termination_time_mins)
-
-            except inductiva.client.ApiException as api_exception:
-                raise api_exception
-
-    def _compute_estimated_price(self, api_instance):
+                    zone=self.zone)
+            self._api.delete_instance_group(body=instance_group_config)
+            termination_time_mins = (time.time() - start_time) / 60
+            logging.info(
+                "Machine group of %s machines successfully"
+                "terminated in %.2f mins.", self.num_machines,
+                termination_time_mins)
+
+        except inductiva.client.ApiException as api_exception:
+            raise api_exception
+
+    def estimate_price(self):
         """Returns an estimated price per hour of a machine group."""
         #TODO: Contemplate disk size in the price.
-        instance_price = api_instance.get_instance_price(
-            body=inductiva.client.model.instance.Instance(id=self.machine_type,
-                                                          zone=self.zone))
+        estimated_price = self._api.get_instance_price({
+            "machine_type": self.machine_type,
+            "zone": self.zone,
+            "spot": self.spot
+        })
+        logging.info("Estimated price per hour: %s $/h", estimated_price.body)
+        return estimated_price.body
+
+    def status(self):
+        """Returns the status of a machine group if it exists.
 
-        if self.spot:
-            estimated_price = instance_price.body[
-                "on_demand"] * self.num_machines
-        else:
-            estimated_price = instance_price.body[
-                "preemptible"] * self.num_machines
+        Otherwise returns None"""
+        response = self._api.get_group_status({"name": self.name})
 
-        return estimated_price
+        if response is None:
+            logging.info("Machine group does not exist.")
```

Comment:
Maybe be explicit that this is using the name to fetch the machine group and log `f"Machine group '{self.name}' does not exist."`

Reply:
`f"Machine group does not exist: {self.name}"` is better as it avoids ambiguities if the name of the machine is something that looks like it could be part of the sentence and is more easily automatically processed. Check in the [styleguide](https://google.github.io/styleguide/pyguide.html#3102-error-messages)